src/main/java/Decoder.java

```diff
@@ -0,0 +1,17 @@
+public class Decoder {
+    public static Command parseInput(String userInput) {
+        Command newCommand;
+        if (userInput.equals("bye")) {
+            newCommand = new ExitProgramCommand();
+        } else if (userInput.equals("list")) {
+            newCommand = new PrintListCommand();
+        } else if (userInput.startsWith("mark")) {
+            newCommand = new UpdateTaskStatusCommand(userInput, true);
+        } else if (userInput.startsWith("unmark")) {
+            newCommand = new UpdateTaskStatusCommand(userInput, false);
+        } else {
+            newCommand = new AddTaskCommand(userInput);
+        }
+        return newCommand;
+    }
```

Comment:
General comment: good conditional formatting and good indenting (in the sense that there are no deeply nested conditionals - the next level of abstraction is generally well-encapsulated in new Object/helper methods/functions